File: 백준/Silver/1676. 팩토리얼 0의 개수/팩토리얼 0의 개수.py
def factorial(n):
    if n <= 1:
        return 1
    return n * factorial(n - 1)

N = int(input())

tg = factorial(N)

cnt = 0
for i in str(tg)[::-1]:
    if i != '0':
        break
    cnt += 1
    
print(cnt)

File: 백준/Silver/1676. 팩토리얼 0의 개수/팩토리얼 0의 개수.java
import java.math.BigInteger;
import java.util.Scanner;

public class Main {

    public static BigInteger factorial(int n) {
        if (n <= 1) {
            return BigInteger.ONE;
        }
        return BigInteger.valueOf(n).multiply(factorial(n - 1));
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();

        BigInteger tg = factorial(N);

        String str = tg.toString();
        int cnt = 0;
        for (int i = str.length() - 1; i >= 0; i--) {
            if (str.charAt(i) != '0') {
                break;
            }
            cnt++;
        }

        System.out.println(cnt);
    }
}
